components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_icon.jsx

```diff
@@ -51,9 +68,21 @@ export default class SidebarChannelButtonOrLinkIcon extends React.PureComponent
                     <ArchiveIcon className='icon icon__archive'/>
                 );
             } else if (this.props.teammateId && this.props.teammateIsBot) {
-                icon = (
-                    <BotIcon className='icon icon__bot'/>
-                );
+                // Use default bot icon
+                icon = (<BotIcon className='icon icon__bot'/>);
+
+                // Attempt to display custom icon if channelIconUrl has changed
+                // or if there was no error when loading custom svg
+                if ((this.props.channelIconUrl && !this.state.svgError) ||
+                    this.props.channelIconUrl !== this.state.channelIconUrl) {
+                    icon = (
+                        <Svg
```

Comment:
What's the experience on load? Will the icon be empty until we fetch it? Is it feasible to render the default icon until it's loaded?

Reply:
Yes, I actually had the default bot showing while loading the image, but the experience was a bit jittery, so it left it as blank for now.

Reply:
Can you elaborate on what jittery means in this context? As in, are we flashing between the icons, or flashing no icon at all?

Reply:
Jittery in the sense that you momentarily see the default icon which is immediately replaced with the custom icon. Happens really fast. As opposed to No icon and then you see the custom icon appear.